### DISCUSSION TITLE
Export of Parameter Identification

### DISCUSSION THREAD

**Discussion by PavelBal** [Open-Systems-Pharmacology/OSPSuite.Core#913] [Ideas]
Starting from the issue https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/issues/906
we propose another approach. Creating a template for R code does not make any sense because we do not have R code for PI (as opposed to Matlab). Instead, we can export a Parameter Identification (PI) as .json files.

## Export PI
- json file with the configuration
- Simulations as pkml
- Observed data as pkml (or as part of json - tbd)
- Export as .zip (check how good R can work with zips)

The good thing is that export of PI configuration is already defined in PK-Sim snapshots:

```
"ParameterIdentifications": [
    {
      "Name": "Parameter Identification 1",
      "Simulations": [
        "Smith1981 iv 5mg Mida"
      ],
      "Configuration": {
        "LLOQMode": "SimulationOutputAsObservedDataLLOQ",
        "RemoveLLOQMode": "Never",
        "CalculateJacobian": false,
        "Algorithm": {
          "Name": "Levenberg - Marquardt (MPFit)",
          "Properties": [
            {
              "Name": "ftol",
              "Value": 0.001
            },
            {
              "Name": "xtol",
              "Value": 1E-06
            },
            {
              "Name": "gtol",
              "Value": 1E-10
            },
            {
              "Name": "stepfactor",
              "Value": 100.0
            },
            {
              "Name": "maxiter",
              "Value": 200
            },
            {
              "Name": "maxfev",
              "Value": 0
            },
            {
              "Name": "epsfcn",
              "Value": 1E-09
            }
          ]
        }
      },
      "OutputMappings": [
        {
          "Scaling": "Log",
          "Path": "Smith1981 iv 5mg Mida|Organism|PeripheralVenousBlood|Midazolam|Plasma (Peripheral Venous Blood)",
          "ObservedData": "WorkShopData.Ex01_Smith1981.IV_5mg_Midazolam_Male"
        }
      ],
      "IdentificationParameters": [
        {
          "Name": "Lipophilicity",
          "Scaling": "Linear",
          "LinkedParameters": [
            "Smith1981 iv 5mg Mida|Midazolam|Lipophilicity"
          ],
          "Parameters": [
            {
              "Name": "Start value",
              "Value": 3.9,
              "Unit": "Log Units"
            },
            {
              "Name": "MinValue",
              "Value": -10.0,
              "Unit": "Log Units"
            },
            {
              "Name": "MaxValue",
              "Value": 10.0,
              "Unit": "Log Units"
            }
          ]
        }
      ]
    }
```

We could discuss of whether to change the structure a bit or just completely re-used the code. Also, export of observed data as part of the json is already in place. Alternatively, observed data could be exported as pkml and referenced in the json-configuration.

Such configurations can be then parsed in R, but this would be a part of the PI-package and can be developed separately (I will set up an initial discussion meeting).

**Comment by PavelBal** [comment]
@msevestre 
I am a bit lost... I thought we had an idea to export the PI as pkml and load the configuration in R, and I thought this has been implemented already, no?

**Comment by msevestre** [comment]
I cannot remember. 
This was not implemented in ospsuite-r. Did you/we do it maybe in ospsuite.parameteridentification? This is where it would belong IMO

**Comment by PavelBal** [comment]
I mean the export to pkml, I thought this was implemented (in dev mode) and wanted to inspect the exported objects from R.. But maybe I just dreamed about this being implemented already :D
